authors/apps/article/models.py

```diff
@@ -240,3 +240,46 @@ class ArticleFavourite(models.Model):
     )
     favourited = models.BooleanField(default=False)
 
+
+class ArticleBookmark(models.Model):
+    """Articles bookmarked Model"""
+    user = models.ForeignKey(
+        User, related_name="bookmarks", on_delete=models.CASCADE)
+    article = models.ForeignKey(
+        Article, related_name='bookmarks', on_delete=models.CASCADE, to_field="slug")
+    created = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def get_bookmark(user_id, slug):
+        """Fetch bookmark"""
+        try:
+            bookmark = ArticleBookmark.objects.get(user=user_id, article=slug)
+            return bookmark
+        except ArticleBookmark.DoesNotExist:
+            pass
+
+        return False
+
+    @staticmethod
+    def create_bookmark(user_id, slug):
+        """create a bookmark"""
+        return_message = {"error": "Bookmark already exists"}
+
+        if not ArticleBookmark.get_bookmark(user_id, slug):
+            ArticleBookmark.objects.create(user=user_id, article=slug)
+            return_message = {"message": "Bookmark created"}
+
+        return return_message
+
+    @staticmethod
+    def remove_bookmark(user_id, slug):
+        """Remoove a bookmark"""
```

Comment:
Nitpick fix: typo on `remoove`
Also, These comments will be unnecessary since they just say the same thing as the function names. Comments should be descriptive of what the function. if they aren't descriptive like in our case, you can use docstrings i.e make the comment start with `#` and not the triple quotes.